x/ibc/04-channel/keeper/querier.go

```diff
@@ -34,3 +34,35 @@ func QuerierChannels(ctx sdk.Context, req abci.RequestQuery, k Keeper) ([]byte,
 
 	return res, nil
 }
+
+// QuerierConnectionChannels defines the sdk.Querier to query all the channels for a connection.
+func QuerierConnectionChannels(ctx sdk.Context, req abci.RequestQuery, k Keeper) ([]byte, error) {
+	var params types.QueryConnectionChannelsParams
+
+	if err := k.cdc.UnmarshalJSON(req.Data, &params); err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
+	}
+
+	channels := k.GetAllChannels(ctx)
+
+	connectionChannels := []types.Channel{}
```

Comment:
do we want to include the port and channel ID too?

Reply:
cc @jackzampolin

Reply:
yes please!

Reply:
What you did for the connection querier should also be done to the channel querier afaik. and this should return the local identifiers for channels and ports

Reply:
OK, added local identifiers for channels & ports to both of the channel queriers.

Reply:
this is 🔥